opennms-webapp/src/main/webapp/WEB-INF/applicationContext-spring-security.xml

```diff
@@ -209,17 +209,28 @@
         <beans:property name="postOnly" value="false"/>
   </beans:bean>
 
-    <beans:bean id="successHandler" class="org.springframework.security.web.authentication.SavedRequestAwareAuthenticationSuccessHandler">
+    <beans:bean id="successHandler" class="org.opennms.web.springframework.security.OpenNMSAuthSuccessHandler">
         <beans:property name="defaultTargetUrl" value="/index.jsp"/>
     </beans:bean>
-    <beans:bean id="failureHandler" class="org.springframework.security.web.authentication.SimpleUrlAuthenticationFailureHandler">
+    <beans:bean id="failureHandler" class="org.opennms.web.springframework.security.OpenNMSAuthFailureHandler">
         <beans:property name="defaultFailureUrl" value="/login.jsp?login_error=1"/>
     </beans:bean>
 
-    <beans:bean id="loginUrlAuthenticationEntryPoint"
-                class="org.springframework.security.web.authentication.LoginUrlAuthenticationEntryPoint">
+    <beans:bean id="loginUrlAuthenticationEntryPoint" class="org.springframework.security.web.authentication.LoginUrlAuthenticationEntryPoint">
         <beans:property name="loginFormUrl" value="/login.jsp" />
+        <!-- Enable forceHttps when deploy with reverse proxy ssl -->
+        <beans:property name="forceHttps" value="true" />
```

Comment:
This is something we need to tell our users when they run OpenNMS behind a TLS proxy, right? Should be the default `false`?

Reply:
May be we build a custom class here as well - would be nice to avoid having to change this file.

Reply:
Good idea

Reply:
@indigo423 I changed to use Jesse's idea to create a new class that extends the Spring class, then we don't need forceHttps property setting anymore. It will automatically detect by opennms.web.base-url=https://%x%c/ setting

Reply:
This is fantastic! 🚀